core/management/commands/update_country_regions.py

```diff
@@ -0,0 +1,532 @@
+#   South Sudan -
+#            due to low quality of the data provided by GeoNames.
+#   Sudan -  the data in GeoNames first needs to be updated and organised to
+#            be suitable for algorithmic processing.
+
+import json
+import re
+
+from django.contrib.gis.geos import LineString, Point
+from django.core.management.base import BaseCommand
+from django.db.models import Q
+from django.utils.termcolors import make_style
+
+import requests
+from django_countries.data import COUNTRIES
+from unidecode import unidecode
+
+from hosting.countries import COUNTRIES_DATA
+from hosting.models import LOCATION_REGION, CountryRegion, Whereabouts
+from maps import SRID
+
+GEONAMES_SOURCE_URL = (
+    'http://api.geonames.org/searchJSON?country={country}&featureCode={code}'
+    '&style=FULL&lang=eo&maxRows=1000&username=pasportaservo'
+)
+GEONAMES_DETAILS_SOURCE_URL = (
+    'http://api.geonames.org/getJSON?geonameId={}&username=pasportaservo'
+)
+COMMERCEGUYS_SOURCE_URL = (
+    'https://raw.githubusercontent.com/commerceguys/addressing'
+    '/master/resources/subdivision/{}.json'
+)
+
+
+class Command(BaseCommand):
+    help = """
+        Updates the administrative units (e.g., states or provinces) for a country.
+        """
+
+    def add_arguments(self, parser):
+        """
+        The command can be given a list of ISO country codes to update,
+        as an argument. A missing argument or the word "ALL" mean that
+        all countries in the local database will be refreshed.
+        """
+        parser.add_argument(
+            'update_countries',
+            nargs='*', choices=tuple(sorted(COUNTRIES.keys())) + ('ALL',), default='ALL',
+            metavar='country code',
+            help="one or more country codes to update, omit to update all countries.")
+        parser.add_argument(
+            '-s', '--start-from',
+            default='AA', choices='ABCDEFGHIJKLMNOPQRSTUVWXYZ',
+            help="when all countries are being updated, begin with the ISO code "
+                 "starting with this letter.")
+        parser.add_argument(
+            '-e', '--end-before',
+            default='ZZ', choices='ABCDEFGHIJKLMNOPQRSTUVWXYZ',
+            help="when all countries are being updated, stop before the ISO code "
+                 "starting with this letter.")
+
+    TOPONYM_REPLACEMENTS = {
+        "EG": "Governorate",
+        "IR": "Province",
+        "MP": "Municipality",
+        "NR": "District",
+        "TW": "County",
+        "VN": ["Province", "City"],
+    }
+
+    def handle(self, *args, **options):
+        self.verbosity = options['verbosity']
+        self.update_countries = update_countries = options['update_countries']
+        if not update_countries or update_countries == 'ALL' or 'ALL' in update_countries:
+            update_countries = tuple(
+                c for c in sorted(COUNTRIES.keys())
+                if c >= options['start_from'] and c < options['end_before']
+            )
```

Comment:
Ĉu `update_countries` povus esti generatoro anstataŭ _tuple_: `update_countries = (`…

Reply:
jes! estas restaĵo de kodo, kiun mi poste ŝanĝis.

Reply:
Bone! Sed mi ne vidas la ŝanĝon en la fandpeton.

Reply:
mi ankoraŭ ne puŝis, mi atendis viajn respondojn.